scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala

```diff
@@ -44,7 +44,7 @@ private class PredictDoFn[T, V](graphBytes: DistCache[Array[Byte]],
                                 fetchOp: Seq[String],
                                 @Nullable config: Array[Byte],
                                 inFn: T => Map[String, Tensor],
-                                outFn: Map[String, Tensor] => V) extends DoFn[T, V] {
+                                outFn: (T, Map[String, Tensor]) => V) extends DoFn[T, V] {
```

Comment:
Another option is to create a custom type or case class like `PredictionResult(original: T, outputOps: Map[String, Tensor])`